### PR TITLE
fix typos

### PR DESCRIPTION
**Pull request by musvaage**
_cf_: https://github.com/pandas-dev/pandas/issues/59640

fixes implemented via shell script

### CODE REVIEW

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @musvaage 

FYI: pandas, and many other open source projects, use https://github.com/codespell-project/codespell to help catch these spelling mistakes. If interested you might want to update their corpus so other projects can benefit.

**Comment by musvaage** [comment]
> update their corpus

you could ping @DimitriPapadopoulos regarding this pull

File: doc/source/user_guide/io.rst [hunk 1]
```diff
@@ -169,7 +169,7 @@ dtype_backend : {"numpy_nullable", "pyarrow"}, defaults to NumPy backed DataFram
   implementation when "numpy_nullable" is set, pyarrow is used for all
   dtypes if "pyarrow" is set.
 
-  The dtype_backends are still experimential.
+  The dtype_backends are still experiential.
```
Comment (mattwang44) [inline doc/source/user_guide/io.rst:172]: should be "experimental"?

Comment (musvaage) [reply doc/source/user_guide/io.rst:172]: https://github.com/pandas-dev/pandas/pull/59669

**Comment by musvaage** [comment]
Thanks @mattwang44

**Comment by DimitriPapadopoulos** [comment]
I have added most of these typos in https://github.com/codespell-project/codespell/pull/3531. Thank you.

Without context,  `experimential` could be changed into `experimental` or `experiential`, so I have added both suggestions.

**Comment by musvaage** [comment]
@DimitriPapadopoulos

pandas/io/pytables.py

reverted  this change which shouldn't have been implemented

```diff
-         """return a tuple of my permutated axes, non_indexable at the front"""
+         """return a tuple of my permuted axes, non_indexable at the front"""
```

**Comment by DimitriPapadopoulos** [comment]
I added `permutate` to the rare dictionary, I will remove it altogether.

**Pull request merged**